GVRf/Framework/jni/objects/mesh.h

```diff
@@ -30,344 +30,322 @@
+		auto it = vec2_vectors_.find(key);
+		if (it != vec2_vectors_.end()) {
+			return it->second;
+		} else {
+			std::string error = "Mesh::getVec2Vector() : " + key + " not found";
+			throw error;
+		}
+	}
+
+	void setVec2Vector(std::string key, const std::vector<glm::vec2>& vector) {
+		vec2_vectors_[key] = vector;
+	}
+
+	std::vector<glm::vec3>& getVec3Vector(std::string key) {
+		auto it = vec3_vectors_.find(key);
+		if (it != vec3_vectors_.end()) {
+			return it->second;
+		} else {
+			std::string error = "Mesh::getVec3Vector() : " + key + " not found";
+			throw error;
+		}
+	}
+
+	const std::vector<glm::vec3>& getVec3Vector(std::string key) const {
+		auto it = vec3_vectors_.find(key);
+		if (it != vec3_vectors_.end()) {
+			return it->second;
+		} else {
+			std::string error = "Mesh::getVec3Vector() : " + key + " not found";
+			throw error;
+		}
+	}
+
+	void setVec3Vector(std::string key, const std::vector<glm::vec3>& vector) {
+		vec3_vectors_[key] = vector;
+	}
+
+	std::vector<glm::vec4>& getVec4Vector(std::string key) {
+		auto it = vec4_vectors_.find(key);
+		if (it != vec4_vectors_.end()) {
+			return it->second;
+		} else {
+			std::string error = "Mesh::getVec4Vector() : " + key + " not found";
+			throw error;
+		}
+	}
+
+	const std::vector<glm::vec4>& getVec4Vector(std::string key) const {
+		auto it = vec4_vectors_.find(key);
+		if (it != vec4_vectors_.end()) {
+			return it->second;
+		} else {
+			std::string error = "Mesh::getVec4Vector() : " + key + " not found";
+			throw error;
+		}
+	}
+
+	void setVec4Vector(std::string key, const std::vector<glm::vec4>& vector) {
+		vec4_vectors_[key] = vector;
+	}
+
+	Mesh* getBoundingBox();
+	const float* getBoundingBoxInfo(); // Xmin, Ymin, Zmin and Xmax, Ymax, Zmax
+
+	// /////////////////////////////////////////////////
+	//  code for vertex attribute location
+
+	void setVertexLoc(GLuint loc) {
+		vertexLoc_ = loc;
+	}
+
+	const GLuint getVertexLoc() const {
+		return vertexLoc_;
+	}
+
+	void setNormalLoc(GLuint loc) {
+		normalLoc_ = loc;
+	}
+
+	const GLuint getNormalLoc() const {
+		return normalLoc_;
+	}
+
+	void setTexCoordLoc(GLuint loc) {
+		texCoordLoc_ = loc;
+	}
+
+	const GLuint getTexCoordLoc() const {
+		return texCoordLoc_;
+	}
+
+	void setVertexAttribLocF(GLuint location, std::string key) {
+		attribute_float_keys_[location] = key;
+	}
+
+	void setVertexAttribLocV2(GLuint location, std::string key) {
+		attribute_vec2_keys_[location] = key;
+	}
+
+	void setVertexAttribLocV3(GLuint location, std::string key) {
+		attribute_vec3_keys_[location] = key;
+	}
+
+	void setVertexAttribLocV4(GLuint location, std::string key) {
+		attribute_vec4_keys_[location] = key;
+	}
+
+	// generate VAO
+	void generateVAO(Material::ShaderType key);
+
+	const GLuint getVAOId(Material::ShaderType key) const {
+		auto iterator = vaoID_map_.find(key);
+		return iterator != vaoID_map_.end() ? iterator->second : 0;
+	}
+
+	void cleanUp() {
```

Comment:
I think if you move this right after `~Mesh()`, the diff will be much easier to read.